### PR TITLE
Rename all identifiers named DeploymentConfig to ApplicationConfig

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
We settled on renaming Deployment Configuration to Application Configuration. For details visit https://github.com/pipe-cd/pipe/issues/2778

This PR changes all identifiers including variable names, comments and file names.
Note that the deployment configuration in the context of ECS isn't related at all.

**Which issue(s) this PR fixes**:

Fixes https://github.com/pipe-cd/pipe/issues/2884

**Does this PR introduce a user-facing change?**:

```release-note
The concept that used to be "Deployment Configuration" is now "Application Confguration"
```

### CODE REVIEW

File: pkg/app/piped/cloudprovider/kubernetes/resource/deployment.go [hunk 1]
```diff
@@ -15,10 +15,10 @@
...
 type Deployment struct {
-	Spec DeploymentSpec
+	Spec ApplicationSpec
 }
 
-type DeploymentSpec struct {
+type ApplicationSpec struct {
 	Replicas int
 	Template PodTemplateSpec
 }
```
Comment (nghialv) [inline pkg/app/piped/cloudprovider/kubernetes/resource/deployment.go:24]: These kinds shouldn't be changed.

File: pkg/app/piped/cloudprovider/kubernetes/resource/statefulset.go [hunk 2]
```diff
@@ -15,7 +15,7 @@
 package resource
 
 type StatefulSet struct {
-	Spec DeploymentSpec
+	Spec ApplicationSpec
```
Comment (nghialv) [inline pkg/app/piped/cloudprovider/kubernetes/resource/statefulset.go:18]: This kind shouldn't be changed.

File: pkg/app/piped/deploysource/deploysource.go [hunk 3]
```diff
@@ -181,11 +181,11 @@ func (p *provider) prepare(ctx context.Context, lw io.Writer) (*DeploySource, er
...
-		RepoDir:                 repoDir,
-		AppDir:                  appDir,
-		Revision:                p.revision,
-		DeploymentConfig:        cfg,
-		GenericDeploymentConfig: gdc,
+		RepoDir:                  repoDir,
+		AppDir:                   appDir,
+		Revision:                 p.revision,
+		ApplicationConfig:        cfg,
+		GenericApplicationConfig: gdc,
```
Comment (nghialv) [inline pkg/app/piped/deploysource/deploysource.go:188]: gdc -> gac

File: pkg/app/piped/executor/cloudrun/deploy.go [hunk 4]
```diff
@@ -136,13 +136,13 @@ func (e *deployExecutor) ensurePromote(ctx context.Context) model.StageStatus {
 		return model.StageStatus_STAGE_FAILURE
 	}
 
-	runningDeployCfg := runningDS.DeploymentConfig.CloudRunDeploymentSpec
-	if runningDeployCfg == nil {
-		e.LogPersister.Error("Malformed deployment configuration in running commit: missing CloudRunDeploymentSpec")
+	runningappCfg := runningDS.ApplicationConfig.CloudRunApplicationSpec
```
Comment (nghialv) [inline pkg/app/piped/executor/cloudrun/deploy.go:139]: runningAppCfg

File: pkg/app/piped/cloudprovider/kubernetes/resource/deployment.go (hunk 1 above)
Comment (nakabonne) [reply pkg/app/piped/cloudprovider/kubernetes/resource/deployment.go:24]: Good catch!

**Comment by nakabonne** [comment]
@nghialv I made some revisions. Would you be able to check them?

**Comment by nghialv** [comment]
Feel good!
/lgtm

**Comment by khanhtc1202** [comment]
Nice work 👏 
/approve

**Pull request merged**